Remove non-standard `KeyboardEvent.altGraphKey`

Ahmad-S792 opened the pull request:
#### f1f9f0045afe3ae23d359d72658b600e4a623ac5
<pre>
Remove non-standard `KeyboardEvent.altGraphKey`

<a href="https://bugs.webkit.org/show_bug.cgi?id=248394">https://bugs.webkit.org/show_bug.cgi?id=248394</a>

Reviewed by Chris Dumez.

This pach aligns WebKit with Gecko / Firefox and Blink / Chromium
by removing non-standard KeyboardEvent.altGraphKey.

It is not in web-specification [1].

[1] <a href="https://www.w3.org/TR/uievents/">https://www.w3.org/TR/uievents/</a>

It was never shipped in Gecko / Firefox while removed in Blink
in 2014 and with commit [2]:

[2] <a href="https://src.chromium.org/viewvc/blink?view=revision&amp">https://src.chromium.org/viewvc/blink?view=revision&amp</a>;revision=179545

&gt; Files modified:
* Source/WebCore/dom/KeyboardEvent.cpp:
(KeyboardEvent::initKeyboardEvent):
* Source/WebCore/dom/KeyboardEvent.h:
* Source/WebCore/dom/KeyboardEvent.idl:
* Source/WebCore/dom/UIEventWithKeyState.cpp:
(UIEventWithKeyState::getModifierState):
(UIEventWithKeyState::setModifierKeys):
* Source/WebCore/dom/UIEventWithKeyState.h:
* Source/WebCore/html/HTMLSelectElementWin.cpp:
(HTMLSelectElement::platformHandleKeydownEvent):
* Source/WebCore/page/EventHandler.cpp:
(EventHandler::defaultPageUpDownEventHandler):
(EventHandler::defaultHomeEndEventHandler):
(EventHandler::defaultArrowEventHandler():
(EventHandler::defaultTabEventHandler):
* Source/WebKit/WebProcess/InjectedBundle/API/gtk/DOM/WebKitDOMKeyboardEvent.cpp:
(webkit_dom_keyboard_event_init_keyboard_event):
(webkit_dom_keyboard_event_get_alt_graph_key):
* Source/WebKit/WebProcess/Plugins/PDF/PDFPluginTextAnnotation.mm:
(PDFPluginTextAnnotation::value):
* Source/WebKit/WebProcess/WebPage/WebPage.cpp:
(WebPage::handleKeyEventByRelinquishingFocusToChrome):
* Source/WebKitLegacy/mac/DOM/DOMKeyboardEvent.mm:
((BOOL)altGraphKey):
((void)initKeyboardEvent):

&gt; Tests Rebaselined / Updated:
* LayoutTests/fast/forms/tabs-with-modifiers.html:
* LayoutTests/fast/forms/tabs-with-modifiers-expected.txt:
* LayoutTests/fast/events/init-events.html:
* LayoutTests/fast/events/init-events-expected.txt:
* LayoutTests/fast/events/init-event-clears-capslock.html:
* LayoutTests/fast/events/init-event-clears-capslock-expected.txt:
* LayoutTests/fast/events/constructors/mouse-event-getModifierState.html:
* LayoutTests/fast/events/constructors/mouse-event-getModifierState-expected.txt:
* LayoutTests/fast/events/constructors/keyboard-event-getModifierState.html:
* LayoutTests/fast/events/constructors/keyboard-event-getModifierState-expected.txt:
* LayoutTests/fast/events/constructors/keyboard-event-constructor.html:
* LayoutTests/fast/events/constructors/keyboard-event-constructor-expected.txt:

Canonical link: <a href="https://commits.webkit.org/273379@main">https://commits.webkit.org/273379@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/95b52b574dfabc24ec3ff67d026a88c1961da68b

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/35274 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/14210 "Built successfully") | [  ~~🛠 mac~~](https://ews-build.webkit.org/#/builders/14/builds/37403 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/38028 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/32/builds/31832 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/36437 "Passed tests") | [  ~~🛠 ios-sim~~](https://ews-build.webkit.org/#/builders/49/builds/16591 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🛠 mac-AS-debug~~](https://ews-build.webkit.org/#/builders/51/builds/11271 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 wpe-wk2~~](https://ews-build.webkit.org/#/builders/34/builds/30710 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/35821 "Passed tests") | [  ~~🧪 ios-wk2~~](https://ews-build.webkit.org/#/builders/49/builds/16591 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 api-mac~~](https://ews-build.webkit.org/#/builders/14/builds/37403 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/10533 "Passed tests") | 
| | [  ~~🧪 ios-wk2-wpt~~](https://ews-build.webkit.org/#/builders/49/builds/16591 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 mac-wk1~~](https://ews-build.webkit.org/#/builders/14/builds/37403 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/39274 "Built successfully") | 
| | [  ~~🧪 api-ios~~](https://ews-build.webkit.org/#/builders/49/builds/16591 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 mac-wk2~~](https://ews-build.webkit.org/#/builders/14/builds/37403 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 gtk-wk2~~](https://ews-build.webkit.org/#/builders/1/builds/36568 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/44/builds/10725 "Built successfully") | [  ~~🧪 mac-AS-debug-wk2~~](https://ews-build.webkit.org/#/builders/51/builds/11271 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 api-gtk~~](https://ews-build.webkit.org/#/builders/21/builds/34584 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | 
| | [  ~~🛠 tv-sim~~](https://ews-build.webkit.org/#/builders/46/builds/12492 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 mac-wk2-stress~~](https://ews-build.webkit.org/#/builders/14/builds/37403 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/8069 "Built successfully and passed tests") | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/43/builds/11249 "Built successfully") | | | 
| | [  ~~🛠 watch-sim~~](https://ews-build.webkit.org/#/builders/45/builds/11549 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | | 
<!--EWS-Status-Bubble-End-->

cdumez reviewed (approved):
LGTM

aprotyas reviewed (commented):
LGTM

mcatanzaro commented on Source/WebKit/WebProcess/InjectedBundle/API/gtk/DOM/WebKitDOMKeyboardEvent.h:
```diff
@@ -105,13 +105,12 @@ webkit_dom_keyboard_event_get_modifier_state(WebKitDOMKeyboardEvent* self, const
 **/
 WEBKIT_DEPRECATED void
-webkit_dom_keyboard_event_init_keyboard_event(WebKitDOMKeyboardEvent* self, const gchar* type, gboolean canBubble, gboolean cancelable, WebKitDOMDOMWindow* view, const gchar* keyIdentifier, gulong location, gboolean ctrlKey, gboolean altKey, gboolean shiftKey, gboolean metaKey, gboolean altGraphKey);
+webkit_dom_keyboard_event_init_keyboard_event(WebKitDOMKeyboardEvent* self, const gchar* type, gboolean canBubble, gboolean cancelable, WebKitDOMDOMWindow* view, const gchar* keyIdentifier, gulong location, gboolean ctrlKey, gboolean altKey, gboolean shiftKey, gboolean metaKey);
```
This is public API and you cannot change it. Instead, keep the parameter and add a call to `UNUSED_PARAM(metaKey)` to document that it's intentionally no longer used (and avoid tripping `-Wunused-parameter`).

mcatanzaro replied:
Actually sorry, as Chris has mentioned in https://github.com/WebKit/WebKit/pull/20404#discussion_r1462564906 we should of course just omit the parameter name from the function signature in the source file rather than use `UNUSED_PARAM()` here.

Ahmad-S792 replied:
OK - so just one change of `false` to `FALSE`?

mcatanzaro replied:
No, you need to keep the `altGraphKey` parameter. That cannot be removed because it is public.

mcatanzaro replied:
(Keep it and just don't use it for anything. Ignore it.)

mcatanzaro commented on Source/WebKit/WebProcess/InjectedBundle/API/gtk/DOM/WebKitDOMKeyboardEvent.cpp:
```diff
@@ -264,9 +264,7 @@ gboolean webkit_dom_keyboard_event_get_alt_graph_key(WebKitDOMKeyboardEvent* sel
-    WebCore::KeyboardEvent* item = WebKit::core(self);
-    gboolean result = item->altGraphKey();
-    return result;
+    return false;
```
```suggestion
    return FALSE;
```

(Return type is gboolean, same as int. Different from bool. bool will convert properly to int in this situation, but we try to be strict anyway.)

Ahmad-S792 replied:
DONE in [b6bfacd](https://github.com/WebKit/WebKit/commit/b6bfacd55a05070624cb47699cb95787edd6bc3b).

mcatanzaro reviewed (commented):
Couple things:

mcatanzaro reviewed (commented):
.

Ahmad-S792 committed:
Remove non-standard `KeyboardEvent.altGraphKey`

https://bugs.webkit.org/show_bug.cgi?id=248394

Reviewed by Chris Dumez.

This pach aligns WebKit with Gecko / Firefox and Blink / Chromium
by removing non-standard KeyboardEvent.altGraphKey.

It is not in web-specification [1].

[1] https://www.w3.org/TR/uievents/

It was never shipped in Gecko / Firefox while removed in Blink
in 2014 and with commit [2]:

[2] https://src.chromium.org/viewvc/blink?view=revision&revision=179545

> Files modified:
* Source/WebCore/dom/KeyboardEvent.cpp:
(KeyboardEvent::initKeyboardEvent):
* Source/WebCore/dom/KeyboardEvent.h:
* Source/WebCore/dom/KeyboardEvent.idl:
* Source/WebCore/dom/UIEventWithKeyState.cpp:
(UIEventWithKeyState::getModifierState):
(UIEventWithKeyState::setModifierKeys):
* Source/WebCore/dom/UIEventWithKeyState.h:
* Source/WebCore/html/HTMLSelectElementWin.cpp:
(HTMLSelectElement::platformHandleKeydownEvent):
* Source/WebCore/page/EventHandler.cpp:
(EventHandler::defaultPageUpDownEventHandler):
(EventHandler::defaultHomeEndEventHandler):
(EventHandler::defaultArrowEventHandler():
(EventHandler::defaultTabEventHandler):
* Source/WebKit/WebProcess/InjectedBundle/API/gtk/DOM/WebKitDOMKeyboardEvent.cpp:
(webkit_dom_keyboard_event_init_keyboard_event):
(webkit_dom_keyboard_event_get_alt_graph_key):
* Source/WebKit/WebProcess/Plugins/PDF/PDFPluginTextAnnotation.mm:
(PDFPluginTextAnnotation::value):
* Source/WebKit/WebProcess/WebPage/WebPage.cpp:
(WebPage::handleKeyEventByRelinquishingFocusToChrome):
* Source/WebKitLegacy/mac/DOM/DOMKeyboardEvent.mm:
((BOOL)altGraphKey):
((void)initKeyboardEvent):

> Tests Rebaselined / Updated:
* LayoutTests/fast/forms/tabs-with-modifiers.html:
* LayoutTests/fast/forms/tabs-with-modifiers-expected.txt:
* LayoutTests/fast/events/init-events.html:
* LayoutTests/fast/events/init-events-expected.txt:
* LayoutTests/fast/events/init-event-clears-capslock.html:
* LayoutTests/fast/events/init-event-clears-capslock-expected.txt:
* LayoutTests/fast/events/constructors/mouse-event-getModifierState.html:
* LayoutTests/fast/events/constructors/mouse-event-getModifierState-expected.txt:
* LayoutTests/fast/events/constructors/keyboard-event-getModifierState.html:
* LayoutTests/fast/events/constructors/keyboard-event-getModifierState-expected.txt:
* LayoutTests/fast/events/constructors/keyboard-event-constructor.html:
* LayoutTests/fast/events/constructors/keyboard-event-constructor-expected.txt:

Canonical link: https://commits.webkit.org/273379@main
